src/wallet/rpcwallet.cpp

```diff
@@ -4162,6 +4183,240 @@ UniValue z_gettotalbalance(const UniValue& params, bool fHelp)
+            "    \"valueZat\": xxxxx,\n                 (numeric) the amount in " + MINOR_CURRENCY_UNIT + "\n"
+            "  },\n"
+            "  \"transparent_watchonly\": {             (object, optional) the balances of transparent funds associated with watch-only transparent addresses\n"
+            "                                         Each such address is treated as being associated with an independent spending authority.\n"
+            "    \"<t_addr_1>\": {...},\n"
+            "    \"<t_addr_2>\": {...},\n"
+            "    ...\n"
+            "  },\n"
+            "  \"sprout\": {                            (object, optional) the balances held by each Sprout spending key in the wallet, keyed by address.\n"
+            "                                         Each such address is treated as being associated with an independent spending authority.\n"
+            "    \"<sprout_addr_1>\": {...},\n"
+            "    \"<sprout_addr_2>\": {...},\n"
+            "    ...\n"
+            "  },\n"
+            "  \"sprout_watchonly_received\": {         (object, optional) the received total visible to each Sprout viewing key in the wallet where the wallet\n"
+            "                                         has no corresponding spending authority, keyed by address. Each such address is treated as being \n"
+            "                                         associated with an independent spending authority. Note that these do not represent balances, as Sprout\n"
+            "                                         viewing keys are not capable of detecting when notes are spent.\n"
+            "    \"<sprout_addr_1>\": {...}\n"
+            "    ...\n"
+            "  },\n"
+            "  \"legacy_sapling\": {                    (object, optional) the balances held by each legacy Sapling spending key in the wallet, keyed by address.\n"
+            "                                           Each such address is treated as being associated with an independent spending authority.\n"
+            "    \"<sapling_addr_1>\": {...},\n"
+            "    \"<sapling_addr_2>\": {...},\n"
+            "    ...\n"
+            "  },\n"
+            "  \"sapling_watchonly\": {                 (object, optional) the balance viewable by each Sapling viewing key in the wallet for which the wallet\n"
+            "                                           does not hold spending authority, keyed by the default address corresponding to the associated incoming\n"
+            "                                           viewing key. Each such address is associated with an independent spending authority. \n"
+            "    \"<sapling_addr_1>\": {...},\n"
+            "    \"<sapling_addr_2>\": {...},\n"
+            "    ...\n"
+            "  },\n"
+            "  \"accounts\": [                          (object, optional) the balances held by each Unified Account spending authority in the wallet\n"
```

Comment:
```suggestion
            "  \"accounts\": [                          (array, optional) the balances held by each Unified Account spending authority in the wallet\n"
```

Reply:
I assume this isn’t a map (object) because the account ids are numeric, not strings. Are account ids sequential? I.e., will they always match their index in the array?

Reply:
Account IDs are not necessarily sequential.